Delivery tracking

alexisjanvier opened the pull request:
Close #79 

![deliveryTracking](https://user-images.githubusercontent.com/547706/78105397-60689280-73f1-11ea-845c-c4cb8c369cac.gif)

alexisjanvier committed:
Migration and remove all MANAGEMENT_BUILT

references

alexisjanvier committed:
Bootsrap delevery tracking view

alexisjanvier committed:
Setup delivery form

tmaziere commented:
"N° de bon de livraison" field is missing. It's an external id.

Commit:
Add business rules on API

Commit:
Handle json types

Commit:
Fix request update

Commit:
Fix deliveryTracking camel case

Commit:
Fix authorization when not updating status

Commit:
Implement mutation on delivery tracking

alexisjanvier committed:
Add form validation

and manage action buttons against status

alexisjanvier committed:
Change the status when everything is delivered

alexisjanvier committed:
delivery percentage counter

floo51 commented on apps/admin/src/request/production-manager/DeliveryTracking.js:
```diff
@@ -0,0 +1,317 @@
+                            (formValues.maskSmallSizeDelivered + formValues.maskLargeSizeDelivered) < 1 ||
+                            Object.keys(formError).length
+                        }
+                        onClick={handleDeliverClick}
```
I think we should use a form and `onSubmit` instead of a button and `onClick`

alexisjanvier replied:
We don't have a form ...

alexisjanvier committed:
Minor improvements

alexisjanvier committed:
Small refactoring

alexisjanvier committed:
Rename migration

to make it work on next deploy
because globa_stats_view_update has already be played

floo51 commented on apps/api/src/request/router.js:
```diff
@@ -46,20 +47,38 @@ router.get('/:id', async ctx => {
+        updatedData.maskLargeSizeDeliveredQuantity = quantitiesDelivered.large;
+
+        if (
+            quantitiesDelivered.small >= request.maskSmallSizeQuantity &&
```
Shouldn't we trigger and error when it's `>`?

alexisjanvier replied:
We discussed it a little with @tmaziere, and in fact we said that if they wanted to deliver more, it had to be possible!
